### PR TITLE
create and push moby bundles as docker image

### PR DESCRIPTION
**Pull request by crazy-max**
Creates a `pkg` job in `ci` workflow to be able to push a multi-platform scratch image that only contains our bundles so we are able to grab them with [undock](https://crazymax.dev/undock/) for example.

I use named contexts feature to add the bundles context in a scratch stage because cross build does not work with buildkit platforms field atm. https://github.com/moby/moby/pull/43529 would fix this in the future.

Tested with:

```
$ DOCKER_CROSSPLATFORMS="linux/amd64 linux/arm/v5 linux/arm/v6 linux/arm/v7 linux/arm64 linux/ppc64le linux/s390x windows/amd64 windows/arm64" docker buildx bake cross
$ docker buildx bake pkg --push --set *.tags=crazymax/moby-bin
```

See https://hub.docker.com/repository/docker/crazymax/moby-bin/tags?page=1&ordering=last_updated

image

```console
$ undock --rm-dist --all crazymax/moby-bin:latest ./moby-bin
...
$ tree -nh ./moby-bin/
[ 512]  ./moby-bin/
├── [ 512]  linux_amd64
│   ├── [ 37M]  containerd
│   ├── [9.2M]  containerd-shim-runc-v2
│   ├── [ 20M]  ctr
│   ├── [720K]  docker-init
│   ├── [1.9M]  docker-proxy
│   ├── [ 60M]  dockerd
│   ├── [ 13K]  dockerd-rootless-setuptool.sh
│   ├── [5.1K]  dockerd-rootless.sh
│   ├── [ 10M]  rootlesskit
│   ├── [6.7M]  rootlesskit-docker-proxy
│   ├── [ 13M]  runc
│   └── [ 32M]  vpnkit
├── [ 512]  linux_arm64
│   ├── [1.9M]  docker-proxy
│   └── [ 57M]  dockerd
├── [ 512]  linux_armv5
│   ├── [2.5M]  docker-proxy
│   └── [ 57M]  dockerd
├── [ 512]  linux_armv6
│   ├── [2.5M]  docker-proxy
│   └── [ 57M]  dockerd
├── [ 512]  linux_armv7
│   ├── [2.4M]  docker-proxy
│   └── [ 57M]  dockerd
├── [ 512]  linux_ppc64le
│   ├── [2.0M]  docker-proxy
│   └── [ 60M]  dockerd
├── [ 512]  linux_s390x
│   ├── [2.7M]  docker-proxy
│   └── [ 62M]  dockerd
├── [ 512]  windows_amd64
│   ├── [1.9M]  docker-proxy.exe
│   └── [ 55M]  dockerd.exe
└── [ 512]  windows_arm64
    ├── [1.8M]  docker-proxy.exe
    └── [ 53M]  dockerd.exe

9 directories, 28 files
```

Signed-off-by: CrazyMax <crazy-max@users.noreply.github.com>

### CODE REVIEW

**Comment by crazy-max** [comment]
Login step to the registry not added yet. Will push that soon.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -95,11 +96,68 @@ jobs:
...
+        with:
+          images: |
+            ${{ env.MOBYBIN_REPO_SLUG }}
+          ### versioning strategy
+          ### push tag v22.06.0-beta.1
+          # dockereng/moby-bin:v22.06.0-beta.1
+          ### push on master
+          # dockereng/moby-bin:master-2fbc307
+          ### push on 22.06 branch
+          # dockereng/moby-bin:22.06-48a1449
```
Comment (crazy-max) [inline .github/workflows/ci.yml:131]: here is the tagging strategy

**Comment by crazy-max** [comment]
ci failure unrelated, github might have some issues

```
#40 0.127 + RUNC_BUILDTAGS=seccomp
#40 0.127 + echo 'Install runc version v1.1.2 (build tags: seccomp)'
#40 0.127 + git clone https://github.com/opencontainers/runc.git /tmp/tmp.1ZFJNnZMfI/src/github.com/opencontainers/runc
#40 0.127 Install runc version v1.1.2 (build tags: seccomp)
#40 0.129 Cloning into '/tmp/tmp.1ZFJNnZMfI/src/github.com/opencontainers/runc'...
#40 1.216 error: RPC failed; curl 56 GnuTLS recv error (-54): Error in the pull function.
#40 1.216 error: 5718 bytes of body are still expected
#40 1.219 fetch-pack: unexpected disconnect while reading sideband packet
#40 1.220 fatal: early EOF
#40 1.220 fatal: index-pack failed
```

**Review by rumpl — APPROVED** [review]

**Comment by crazy-max** [comment]
One thing I forgot to mention is it will create statically linked binaries but in Docker Desktop we use the Debian package which has dyn linked binaries. This might be ok for testing but I would suggest that we move forward with https://github.com/moby/moby/pull/43529 to be able to cross comp dynamically linked binaries against any arch.

Other thing is `docker-init` is missing for other architectures but it's solved with https://github.com/moby/moby/pull/43529

**Pull request closed without merging**